core/services/workflows/model.go

```diff
@@ -0,0 +1,86 @@
+package workflows
+
+import (
+	"fmt"
+	"sigs.k8s.io/yaml"
+)
+
+func ParseWorkflowModelFromString(s string) (WorkflowModel, error) {
+    var model WorkflowModel
+    err := yaml.Unmarshal([]byte(s), &model)
+    if err != nil {
+        // TODO: Use chainlink/core/logger to log the error
```

Comment:
Hmmm... I wouldn't perform any logging here. Logging an error feels to me like it's something you'd do at the orchestration layer, and parsing feels like it's lower down than that. I'd suggest moving any logging (including TODOs) higher up.